File: javascript/word-count/word-count.js
var Words = function() {

  var cleanMsg = function(msg) {
    return msg.toLowerCase()
    .replace(/[ \t\r\n]+/g, ' ')
    .replace(/^ /g, '')
    .replace(/ $/g, '')
  }

  this.count = function(msg) {
    return cleanMsg(msg).split(' ').reduce(function(accum, word) {
      if (!(word in accum) || isNaN(accum[word])) { accum[word] = 0 }
      accum[word]++;
      return accum;
    }, {})
  };
};

module.exports = Words;

File: python/word-count/word_count.py
# -*- coding: utf-8 -*-

import re
import string

def word_count(msg):
    msg = re.sub(u'[^\w\u0370-\u1FFF+\d]|_', ' ', msg.decode("utf-8").lower())
    words = msg.split()
    counted = dict((word,0) for word in words)
    for word in words:
        counted[word] += 1
    return counted
